src/pages/upload/index.tsx

```diff
@@ -74,88 +86,206 @@ const fileUploadReducer = (state: any, action: any) => {
+            >
+              Topic
+            </label>
+            <p className="mt-2 text-center text-sm text-gray-600 border border-gray-300 rounded-md p-2">
+              coming soon...
+            </p>
+          </div>
+          <div className="space-y-1">
+            <label
+              htmlFor="instructorId"
+              className="block text-sm font-medium text-gray-700"
+            >
+              Instructor
+            </label>
+            <select
+              id="instructorId"
+              name="instructorId"
+              className="mt-1 block w-full pl-3 pr-10 py-2 text-base border-gray-300 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm rounded-md"
+              value={courseInstructorId}
+              onChange={(e) => {
+                setCourseInstructorId(e.target.value)
+              }}
+            >
+              {instructors.map(
+                ({
+                  eggheadInstructorId,
+                  person,
+                }: {
+                  eggheadInstructorId: string
+                  person: {name: string}
+                }) => {
+                  return (
+                    <option value={eggheadInstructorId}>
+                      {person['name']}
+                    </option>
+                  )
+                },
+              )}
+            </select>
+          </div>
+          <div className="space-y-1">
+            <label className="block text-sm font-medium text-gray-700">
+              Video Files
+            </label>
+            {/* Drop zone UI adapted from https://larainfo.com/blogs/tailwind-css-drag-and-drop-file-upload-ui */}
+            <div className="max-w-xl">
+              <label className="flex justify-center w-full h-32 px-4 transition bg-white border-2 border-gray-300 border-dashed rounded-md appearance-none cursor-pointer hover:border-gray-400 focus:outline-none">
+                <span className="flex items-center space-x-2">
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    className="w-6 h-6 text-gray-600"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    stroke="currentColor"
+                    stroke-width="2"
+                  >
+                    <path
+                      stroke-linecap="round"
+                      stroke-linejoin="round"
+                      d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12"
+                    />
+                  </svg>
+                  <span className="font-medium text-gray-600">
+                    Drop video files, or{' '}
+                    <span className="text-blue-600 underline">browse</span>
+                  </span>
+                </span>
+                <VideoUploader
+                  dispatch={dispatch}
+                  setLessonMetadata={setLessonMetadata}
+                />
```

Comment:
Browse to upload works, but I didn't get the drag and drop working. I decided not to go down that rabbit hole at the moment. The intention here is more to cast a vision of what this Course Builder can be.